jx_sqlite/insert_table.py

```diff
@@ -281,6 +282,28 @@ def _flatten(data, uid, parent_id, order, full_path, nested_path, row=None, guid
                 elif c.type=="nested" and value_type=="object":
                     value_type = "nested"
                     v = [v]
+                elif c.es_index!=table:
+                    from_doc = doc_collection.get(c.nested_path[0], None)
```

Comment:
you are attempting to detect a situation, when running `test_union_nested_objects`, on the column `_c.v` changes from a fact_table primitive, to a property in a nested object array.  You attempt to detect this by noticing the es_index is not the same as the expected table.  You should be checking if the nesting level of the current column (`c`) is less than the depth of the current `_flatten` call.

    len(c.nested_path) < len(nested_path)